### PR TITLE
refactor: reduce duplication and nil-pointer issues in server pkg

### PR DESCRIPTION
**Pull request by spy16**
Server package had lot of duplication, nil-pointer issues, etc. 

This refactor tries to maximise re-use (for example, instead of hand-coding the request/response models, now using the swagger generated models directly — so it will always be in sync) and also added validations wherever necessary to prevent nil-pointer issues.

Also moved to `go-chi/v5` router (since gorilla mux is now archived)

### CODE REVIEW

File: cli/config/configs.go [hunk 1]
```diff
@@ -16,7 +16,8 @@ var ErrClientConfigNotFound = errors.New(heredoc.Doc(`
 `))
 
 type ClientConfig struct {
-	Host string `yaml:"host" cmdx:"host"`
+	Host     string `yaml:"host" cmdx:"host"`
+	UseHTTPS bool   `yaml:"use_https" cmdx:"use_https"`
```
Comment (mabdh) [inline cli/config/configs.go:20]: what if we call it `TLS` instead of `HTTPS`

File: internal/server/v1/alert/vars.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
+package alert
+
+import "golang.org/x/exp/maps"
+
+var SuppliedVariables = []string{"name", "team", "entity"}
```
Comment (mabdh) [inline internal/server/v1/alert/vars.go:5]: I am not sure with this but what are these supplied variables for?

Comment (spy16) [reply internal/server/v1/alert/vars.go:5]: These are variables that are auto injected from dex. I'm not sure of their exact function as well honestly. This bit was done by @rohil .. in the refactor i retained the behavior and just moved things around.

File: cli/config/configs.go (hunk 1 above)
Comment (spy16) [reply cli/config/configs.go:20]: this reflects the schema whitelist in the swagger client. also, user is expected to set `host` config without `http://` or `https://` scheme prefix. so kept this as `use_https` to make it more obvious that it's going to use `https://` or not for calls.

**Review by mabdh — APPROVED** [review]

**Pull request merged**